### PR TITLE
DM-13670 add tweakPhotometryErrors and some new plots

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: include/lsst/jointcal/PhotometryModel.h [hunk 1]
```diff
@@ -142,9 +139,32 @@ class PhotometryModel {
...
+
+    /// Add a small magnitude offset to the "instrumental magnitude" error, in quadrature.
+    double tweakMagnitudeError(jointcal::MeasuredStar const &measuredStar) const {
+        if (errorPedestal == 0) {
+            return measuredStar.getInstMagErr();
+        } else {
+            return std::hypot(measuredStar.getInstMagErr(), errorPedestal);
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (parejkoj) [reply include/lsst/jointcal/PhotometryModel.h]: I'm not sure I understand your question: for the same `errorPedestal` (for values less than `~0.10`), these two calculations should result in the nearly the same additional error.

File: tests/test_jointcal_cfht.py [hunk 2]
```diff
@@ -209,6 +209,33 @@ def test_jointcalTask_2_visits_constrainedMagnitude_no_astrometry(self):
...
+
+        # We're allowing more error in the fit, so PA1 may be worse.
+        pa1 = 0.018
+        # Final chi2 is much lower, because all sources contribute more error.
+        metrics['photometry_final_chi2'] = 1800.08
+        # ndof shouldn't change much; slightly different likelihood contours
+        metrics['photometry_final_ndof'] = 1338
+
+        self._testJointcalTask(2, None, None, pa1, metrics=metrics)
+
```
[the remark this answers is not in the document]
Comment (parejkoj) [reply tests/test_jointcal_cfht.py]: The HSC test data I have has two few chips to be of use with the constrained model.

**Comment by parejkoj** [comment]
Note that ci_hsc doesn't use jointcal at all, so it's irrelevant for testing this.

File: notebooks/Outlier plots.ipynb [hunk 3]
```diff
@@ -46,19 +46,31 @@
...
+   "execution_count": 33,
    "metadata": {},
    "outputs": [],
    "source": [
     "outlierRe = re.compile(r\"\\(Measured \\+ Reference \\= Total\\): ([0-9]*) \\+ ([0-9]*) \\= ([0-9]*)\")\n",
     "chi2Re = re.compile(r\"\")\n",
     "\n",
+    "xRe = r\"x: (?P<x>[+-]?\\d+(?:.\\d*)?)\"\n",
+    "yRe = r\"y: (?P<y>[+-]?\\d+(?:.\\d*)?)\"\n",
+    "fluxRe = r\"flux: (?P<flux>([+-]?(?:0|[1-9]\\d*)(?:\\.\\d*)?(?:[eE][+\\-]?\\d+))|(\\d+(?:.\\d*)?))\"\n",
```
[the remark this answers is not in the document]
Comment (parejkoj) [reply notebooks/Outlier plots.ipynb]: I've added a comment about what it does.

File: include/lsst/jointcal/PhotometryModel.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (taranu) [reply include/lsst/jointcal/PhotometryModel.h]: I meant to say that adding log(x) and log(c) in quadrature is not the same as adding x and c in quadrature, but yes it makes little different for small constants and I don't have an alternative to suggest, so never mind.

**Pull request merged**